### PR TITLE
Make KeyCodes Ord

### PR DESCRIPTION
**Pull request by StarStarJ**
I'd like to be able to use the KeyCodes as a reproducable chain, so sorting would be nice to have

- [ ] Tested on all platforms changed
- [ ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

**Comment by kchibisov** [comment]
Hm, the issue is that I'm not sure how much sense it makes, could you give a more concrete example, as in rust code for that?

It's a bit weird to sort keycodes when we don't have any numbers directly attached to them and we can reorder them at any point in time.

**Comment by StarStarJ** [comment]
I basically have a BTreeSet of pressed keycodes

And a structure that represents a list of keys one has to press in order to trigger a specific event

so lets say the structure contains:
A => DoThat

B,A => DoThis
B has a potential list of follow up keycodes

C,B,A => DoAnother
C has a potential list of follow up keycodes, b as well

C,D,A => DoYetAnother
D is another keyboard in the list of C's follow up keycodes

The actual order does not matter, because the structure and the input keycodes are both sorted.
The key presses are evaluated every frame

I hope this example is good enough for you. I'm not a native english speaker, so it's not so easy to explain it for me

**Comment by kchibisov** [comment]
Ah, so you basically require any ordering when pressing, not specific, since commonly you'd do a `Vec<KeyCode>` and then `index` to track current progress if you want to press, e.g. `gg` or `abc`.

Maybe it's fine to have `Ord` since it's for native keys.

**Review by kchibisov — APPROVED** [review]

**Comment by kchibisov** [comment]
Could you add a `CHANGELOG` entry saying the types you've changed implement `Ord`/`PartialOrd`?

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ And please only add new entries to the top of this list, right below the `# Unre
 
 # Unreleased
 
+- Add `PartialOrd` and `Ord` to `KeyCode` and `NativeKeyCode` to allow them to be compared, e.g. to store them into a `BTreeSet`
```
Comment (kchibisov) [inline CHANGELOG.md:11]: No need to have that many details.

```suggestion
- Implement `PartialOrd` and `Ord` for `KeyCode` and `NativeKeyCode`.
```

**Review by kchibisov — APPROVED** [review]

**Comment by kchibisov** [comment]
Thanks!

**Pull request merged**